flaml/automl/automl.py

```diff
@@ -2081,8 +2117,13 @@ def trainable(self) -> Callable[[dict], Optional[float]]:
         states = self._search_states
         mem_res = self._mem_thres
 
-        def train(config: dict, state):
-
+        def train(config: dict, state, is_report=True):
+            # handle spark broadcast variables
+            if spark_import:
+                if isinstance(state, pyspark.broadcast.Broadcast):
+                    state = state.value
+                if isinstance(is_report, pyspark.broadcast.Broadcast):
```

Comment:
Why would `is_report` ever by this type? Seems it should always be boolean. (Also see comment on the naming of this above)